modules/gathering-bootstrap-diagnostic-data.adoc

```diff
@@ -0,0 +1,69 @@
+// Module included in the following assemblies:
+//
+// * support/troubleshooting/troubleshooting-installations.adoc
+
+[id="gathering-bootstrap-diagnostic-data_{context}"]
+= Gathering bootstrap node diagnostic data
+
+When experiencing bootstrap-related issues, you can gather `bootkube.service` journald unit logs and container logs from the bootstrap node.
+
+.Prerequisites
+
+* The `ssh-agent` process is active on your computer, and you provided both the `ssh-agent` process and the installation program with the same SSH key.
+* If you tried to install a cluster on infrastructure that you provisioned, you must have the fully qualified domain names of the bootstrap node.
+
+.Procedure
+
+. If you have access to the bootstrap node's console, monitor the console until the node reaches the login prompt.
+
+. Verify the Ignition file configuration.
++
+* If you are hosting Ignition configuration files by using an HTTP server.
++
+.. Verify the bootstrap node Ignition file URL:
++
+----
+$ curl -I http://<fully_qualified_domain_name>:<port>/bootstrap.ign  <1>
+----
+<1> The `-I` option returns the header only. If the Ignition file is available on the specified URL, the command returns `200 OK` status. If it is not available, the command returns `404 file not found`.
++
+.. To verify that the Ignition file was received by the bootstrap node, query the HTTP server logs on the serving host. For example, if you are using an Apache web server to serve Ignition files, enter the following command:
++
+----
+$ grep -is 'bootstrap.ign' /var/log/httpd/access_log
+----
++
+If the bootstrap Ignition file is received, the associated `HTTP GET` log message will include a `200 OK` success status, indicating that the request succeeded.
++
+.. If the Ignition file was not received, check that the Ignition files exist and that they have the appropriate file and web server permissions on the serving host directly.
++
+* If you are using a cloud provider mechanism to inject Ignition configuration files into hosts as part of their initial deployment.
++
+.. Review the bootstrap node's console to determine if the mechanism is injecting the bootstrap node Ignition file correctly.
+
+. Verify the availability of the bootstrap node's assigned storage device.
+
+. Verify that the bootstrap node has been assigned an IP address from the DHCP server.
+
+. Collect `bootkube.service` journald unit logs from the bootstrap node. Replace `<cluster_name>.<sub_domain>.<domain>` with appropriate values:
++
+----
+$ ssh core@bootstrap.<cluster_name>.<sub_domain>.<domain> journalctl -b -f -u bootkube.service
```

Comment:
The virtual machines in cluster on cloud platform are private ip address. if user execute this command on local machine, the login will be failed.

Reply:
I have now resolved this as per our Slack conversation, by changing the prerequisite in multiple places throughout the PR to:

* You have SSH access to your hosts.

Because there are multiple ways to access cluster nodes depending on the scenario, this steps back to a higher level and as such, accommodates local host and bastion host situations with a broad brush.